servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/FromInputStreamPublisher.java

```diff
@@ -62,14 +63,27 @@
     private volatile int subscribed;
 
     private final InputStream stream;
+    private final int readChunkSize;
 
     /**
      * A new instance.
      *
      * @param stream the {@link InputStream} to expose as a {@link Publisher}
      */
     FromInputStreamPublisher(final InputStream stream) {
+        this(stream, DEFAULT_READ_CHUNK_SIZE);
+    }
+
+    /**
+     * A new instance.
+     *
+     * @param stream the {@link InputStream} to expose as a {@link Publisher}
+     * @param readChunkSize the maximum length of {@code byte[]} chunks which will be read from the {@link InputStream}
+     * and emitted by the {@link Publisher}.
+     */
+    FromInputStreamPublisher(final InputStream stream, final int readChunkSize) {
         this.stream = requireNonNull(stream);
+        this.readChunkSize = readChunkSize;
```

Comment:
Can you please add validation for this param to make sure it's greater than 0? We usually use this consistent format:
```suggestion
        if (readChunkSize <= 0) {
            throw new IllegalArgumentException("readChunkSize: " + readChunkSize + " (expected: >0)");
        }
```

Reply:
Looks like I can merge the suggestion from the GH UI, let me try :)